cloudinit/tests/test_features.py

```diff
@@ -0,0 +1,59 @@
+# This file is part of cloud-init. See LICENSE file for license information.
+# pylint: disable=no-member,no-name-in-module
+"""
+This file is for testing the feature flag functionality itself,
+NOT for testing any individual feature flag
+"""
+import pytest
+from importlib import reload
+from pathlib import Path
+
+import cloudinit
+import cloudinit.features
+
+
+@pytest.yield_fixture()
+def create_override(request):
+    override_path = Path(cloudinit.__file__).parent / 'feature_overrides.py'
+    with override_path.open('w') as f:
+        for key, value in request.param.items():
+            f.write('{} = {}\n'.format(key, value))
+
+    # The goal is to reload features.py so we also get the
+    # features_overrides.py stuff. This could normally be achieved with
+    # a single reload(cloudinit.features), however, since this fixture will be
+    # called multiple times in the same interpreter, calling
+    # reload(cloudinit.features) a second time won't refresh what already
+    # exists in feature_overrides.py because that's now already defined, so we
+    # need to call reload(cloudinit.feature_overrides) first. But the first
+    # time we call this fixture, cloudinit.feature_overrides doesn't exist yet,
+    # so we need to reload cloudinit.features first before we can reload
+    # cloudinit.feature_overrides, hence the three calls here.
+    reload(cloudinit.features)
+    reload(cloudinit.feature_overrides)
+    reload(cloudinit.features)
+
+    yield
+
+    override_path.unlink()
+
+
+class TestFeatures:
```

Comment:
These tests look good, thanks! I'd also like to see a test that ensures that invalid Python in `feature_overrides.py` will be handled gracefully; at the moment I think we would fail because `import` will raise a `SyntaxError` (not an `ImportError`) in such a case.

(A case where we could end up in a situation with an invalid `feature_overrides.py` which isn't wildly unlikely: if someone writes `feature_overrides.py` and uses a Python language feature that isn't supported in the target platform's Python. For example, `ERROR_ON_USER_DATA_FAILURE: bool = False` raises a `SyntaxError` when imported with Python 3.5, because of the variable annotation.)

Reply:
What does it mean to handle these errors gracefully? I would think we would want to raise the exception so that it can easily be seen/found/fixed in tests, rather than expecting some functionality to work but having it silently not.